modules/core/04-channel/keeper/upgrade.go

```diff
@@ -359,6 +359,35 @@ func (k Keeper) ChanUpgradeAck(
 	return nil
 }
 
+// writeUpgradeTimeoutChannel restores the channel state of an initialising chain in the event that the counterparty chain has passed the timeout set in ChanUpgradeInit to the state before the upgrade was proposed.
+// Auxiliary upgrade state is also deleted.
+// An event is emitted for the handshake step.
+func (k Keeper) writeUpgradeTimeoutChannel(
+	ctx sdk.Context,
+	portID, channelID string,
+) error {
+	defer telemetry.IncrCounter(1, "ibc", "channel", "upgrade-timeout")
+
+	channel, found := k.GetChannel(ctx, portID, channelID)
+	if !found {
+		panic(fmt.Sprintf("could not find existing channel when updating channel state in successful ChanUpgradeTimeout step, channelID: %s, portID: %s", channelID, portID))
+	}
+
+	upgrade, found := k.GetUpgrade(ctx, portID, channelID)
+	if !found {
+		panic(fmt.Sprintf("could not find existing upgrade when cancelling channel upgrade, channelID: %s, portID: %s", channelID, portID))
+	}
+
+	if err := k.AbortUpgrade(ctx, portID, channelID, types.NewUpgradeError(channel.UpgradeSequence, types.ErrUpgradeTimeout)); err != nil {
+		return errorsmod.Wrapf(types.ErrUpgradeRestoreFailed, "err: %v", err)
+	}
```

Comment:
The `AbortUpgrade` function writes an error receipt to state for counterparty upgrade cancellation but in this scenario the counterparty is in its original state and we are just wanting to restore our channel end to its original state also. 

Its only possible to timeout an upgrade handshake if the counterparty stalls on TRY (remaining in an open state) and we have committed the INIT step which we need to revert/timeout. So writing the error receipt here is redundant.

Reply:
https://github.com/cosmos/ibc/issues/986

Reply:
Timeouts can also occur if an insufficient timeout was set during INIT